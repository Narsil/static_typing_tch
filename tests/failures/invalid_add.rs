use static_typing_tch::tensor_check_fn;
use tch::{kind::Kind, Device, Tensor};

#[tensor_check_fn(a, b)]
fn add() {
    // The dimensions do not match
    let a = Tensor::ones(&[2, 3], (Kind::Float, Device::Cpu));
    let b = Tensor::ones(&[3, 2], (Kind::Float, Device::Cpu));

    let _c = a + b;
}

fn main() {
    add();
}
