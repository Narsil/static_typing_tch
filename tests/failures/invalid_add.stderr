error[E0432]: unresolved import `static_typing_tch::tensor_check_fn`
 --> tests/failures/invalid_add.rs:1:5
  |
1 | use static_typing_tch::tensor_check_fn;
  |     ^^^^^^^^^^^^^^^^^^^---------------
  |     |                  |
  |     |                  help: a similar name exists in the module: `tensor_check`
  |     no `tensor_check_fn` in the root

error: cannot determine resolution for the attribute macro `tensor_check_fn`
 --> tests/failures/invalid_add.rs:4:3
  |
4 | #[tensor_check_fn(a, b)]
  |   ^^^^^^^^^^^^^^^
  |
  = note: import resolution is stuck, try simplifying macro imports
